janus_server/src/aggregator.rs

```diff
@@ -3915,10 +4084,115 @@ mod tests {
         assert_eq!(path_segments.next(), Some("collect_jobs"));
-        assert_matches!(path_segments.next(), Some(uuid) => {
-            Uuid::parse_str(uuid).unwrap();
-        });
+        let collect_job_id = Uuid::parse_str(path_segments.next().unwrap()).unwrap();
         assert!(path_segments.next().is_none());
+
+        let collect_job_response = warp::test::request()
+            .method("GET")
+            .path(&format!("/collect_jobs/{}", collect_job_id))
+            .filter(&filter)
+            .await
+            .unwrap()
+            .into_response();
+        assert_eq!(collect_job_response.status(), StatusCode::ACCEPTED);
+
+        datastore
+            .run_tx(|tx| {
+                let collector_hpke_config = task.collector_hpke_config.clone();
+                Box::pin(async move {
+                    let encrypted_helper_aggregate_share = hpke::seal(
+                        &collector_hpke_config,
+                        &HpkeApplicationInfo::new(
+                            task.id,
+                            Label::AggregateShare,
+                            Role::Helper,
+                            Role::Collector,
+                        ),
+                        &[],
+                        //&[0, 1, 2, 3, 4, 5],
+                        &batch_interval.get_encoded(),
+                    )
+                    .unwrap();
+
+                    tx.update_collect_job::<fake::Vdaf, _>(
+                        collect_job_id,
+                        &encrypted_helper_aggregate_share,
+                        &fake::AggregateShare(),
+                        10,
+                        [1; 32],
+                    )
+                    .await
+                    .unwrap();
+
+                    Ok(())
+                })
+            })
+            .await
+            .unwrap();
+
+        let (parts, body) = warp::test::request()
+            .method("GET")
+            .path(&format!("/collect_jobs/{}", collect_job_id))
+            .filter(&filter)
+            .await
+            .unwrap()
+            .into_response()
+            .into_parts();
+
+        assert_eq!(parts.status, StatusCode::OK);
+        let body_bytes = hyper::body::to_bytes(body).await.unwrap();
+
+        let collect_resp = CollectResp::get_decoded(body_bytes.as_ref()).unwrap();
+        assert_eq!(collect_resp.encrypted_agg_shares.len(), 2);
+
+        let leader_aggregate_share = hpke::open(
+            &task.collector_hpke_config,
+            &collector_hpke_recipient,
+            &HpkeApplicationInfo::new(
+                task_id,
+                Label::AggregateShare,
+                Role::Leader,
+                Role::Collector,
+            ),
+            &collect_resp.encrypted_agg_shares[0],
+            &batch_interval.get_encoded(),
+        )
+        .unwrap();
+        fake::AggregateShare::try_from(leader_aggregate_share.as_ref()).unwrap();
```

Comment:
Note that this isn't testing much right now, since `fake::AggregateShare` is just a newtype of the unit type.

We could use/adapt `run_vdaf` to produce values that would let us do an aggregation on a real VDAF, and check that the aggregation result is as expected. Up to you, I won't hold the PR up over this if you think it's not worthwhile.

Reply:
It was easy enough to adapt this test to use `Prio3Aes128Count` and then synthesize fake `AggregateShares` to verify that they get encoded and encrypted appropriately.